src/onnx/parse_qlinearadd.cpp

```diff
@@ -0,0 +1,193 @@
+  per-tensor/layer quantization.
+
+  Outputs
+  C : T
+  Result, has same element type as two inputs
+
+  Type Constraints
+  T : tensor(uint8), tensor(int8)
+  Constrain input and output types to 8 bit signed and unsigned tensors.
+
+*/
+
+struct parse_qlinearadd : op_parser<parse_qlinearadd>
+{
+    std::vector<op_desc> operators() const { return {{"QLinearAdd"}}; }
+
+    // basic type checking for QLinearAdd Operator
+    void check_inputs(const std::vector<instruction_ref>& args) const
+    {
+        if(args.size() < 7)
+            MIGRAPHX_THROW("QLINEARADD: missing inputs");
+
+        const auto& in_a = args[0];
+        const auto& in_b = args[3];
+
+        auto sh_a = in_a->get_shape();
+        auto sh_b = in_b->get_shape();
+        if(sh_a != sh_b)
+            MIGRAPHX_THROW("QLINEARADD: mismatched input shapes");
+
+        auto type_a = sh_a.type();
+        auto type_b = sh_b.type();
+        if(type_a != migraphx::shape::int8_type and type_a != migraphx::shape::uint8_type)
+            MIGRAPHX_THROW("QLINEARADD: unsupported input type");
+        if(type_b != migraphx::shape::int8_type and type_b != migraphx::shape::uint8_type)
+            MIGRAPHX_THROW("QLINEARADD: unsupported input type");
+    }
+
+    instruction_ref bcast_scalar_instr(const migraphx::shape& shape_out,
+                                       const instruction_ref arg_in,
+                                       const onnx_parser::node_info& info) const
+    {
+        auto bcast_instr_out = info.add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", shape_out.lens()}}), arg_in);
+        return bcast_instr_out;
+    }
+
+    // This method is to prep for quantizelinear or dequantizelinear operation for
+    // either the broadcasting of weight-scale or zero-points of qlinearadd operator
+    // outputs: operator op (inputs x, broadcasted: scale (float) & zero_pt (8-bit))
+    instruction_ref bcast_qdq_instr(const std::string& op_name,
+                                    const instruction_ref x_in,
+                                    const instruction_ref arg_fscale,
+                                    const instruction_ref arg_z_pt,
+                                    const onnx_parser::node_info& info) const
+    {
+        auto in_lens = x_in->get_shape().lens();
+
+        // prep 1: broadcast scale. it can come as a scalar or a 1-D tensor.
+        instruction_ref bcast_scale;
+        if(arg_fscale->get_shape().elements() > 1)
+            bcast_scale = info.add_instruction(
+                migraphx::make_op("broadcast", {{"axis", 0}, {"out_lens", in_lens}}), arg_fscale);
+        else
+            bcast_scale = info.add_instruction(
+                migraphx::make_op("multibroadcast", {{"out_lens", in_lens}}), arg_fscale);
+
+        // prep 2: broadcast zero point. it can come as a scalar or a 1-D tensor.
+        instruction_ref bcast_zero_pt;
+        if(arg_z_pt->get_shape().elements() > 1)
```

Comment:
Does this work `get_shape().scalar()`  instead ?

Reply:
The current style one is lot clearer to me, and therefore sticking to that one. 

scalar() definition: in its include file is referring to its strides of 0  -- and I am not sure of its functional relation to elements(), even though its implementation might mean the equivalence. Thanks.

Reply:
Its not equivalent, so its better to use `get_shape().elements()`.